File: src/main/java/com/brandonitaly/bedrockskins/mixins/MixinPlayerEntityRenderState.java
package com.brandonitaly.bedrockskins.mixins;

import com.brandonitaly.bedrockskins.client.BedrockSkinState;
import com.brandonitaly.bedrockskins.pack.SkinId;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.Unique;
import java.util.UUID;
import net.minecraft.client.renderer.entity.state.AvatarRenderState;

@Mixin(AvatarRenderState.class)
public class MixinPlayerEntityRenderState implements BedrockSkinState {
    @Unique
    private SkinId bedrockSkinId;
    @Unique
    private UUID uniqueId;

    @Override
    public SkinId getBedrockSkinId() {
        return bedrockSkinId;
    }

    @Override
    public void setBedrockSkinId(SkinId id) {
        this.bedrockSkinId = id;
    }

    @Override
    public UUID getUniqueId() {
        return uniqueId;
    }

    @Override
    public void setUniqueId(UUID uuid) {
        this.uniqueId = uuid;
    }
}

File: src/main/kotlin/com/brandonitaly/bedrockskins/mixins/MixinPlayerEntityRenderState.kt
package com.brandonitaly.bedrockskins.mixins

import com.brandonitaly.bedrockskins.client.BedrockSkinState
import net.minecraft.client.render.entity.state.PlayerEntityRenderState
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.Unique
import java.util.UUID

@Mixin(PlayerEntityRenderState::class)
class MixinPlayerEntityRenderState : BedrockSkinState {
    @Unique
    override var bedrockSkinKey: String? = null
    @Unique
    override var uniqueId: UUID? = null
}
